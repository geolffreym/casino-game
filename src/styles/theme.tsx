// Global styles can be moved to a separate file for ease of maintenance.
export const globalOverrides = {
  overrides: {
    MuiCssBaseline: {
      '@global': {
        html: {
          margin: 0,
          padding: 0,
          height: '100%'
        },
        body: {
          margin: 0,
          padding: 0,
          height: '100%',
          fontSize: '1rem',
          backgroundColor: '#fff'
        }
      }
    }
  }
}

export const typography = {
  typography: {
    fontFamily: [
      'Tamil Sangam MN',
      'Oswald',
      'Arial',
      'sans-serif'
    ].join(',')
  }
}

export const defaultTheme = {
  palette: {
    primary: {
      light: '#bf1f34',
      main: '#990015',
      dark: '#750212',
      contrastText: '#E4F1F0'
    },
    secondary: {
      main: '#2795B7',
      contrastText: '#fff'
    },
    warning: {
      light: '#FEF5B2',
      main: '#C79C36',
      dark: '#814E22'
    },
    success: {
      light: '#DEF7EC',
      main: '#3C763D',
      dark: '#2D6C59'
    },
    default: {
      main: '#fff',
      contrastText: '#9999'
    },
    grey: {
      50: '#F2F6FF',
      150: '#F9FAFC',
      200: '#E6EBF1',
      250: '#98A5BA'
    }
  }
}
