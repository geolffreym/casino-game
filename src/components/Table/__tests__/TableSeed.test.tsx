import {
  TableWrapper, TableBody, TableColumn,
  TableScroller, TableHeader
} from '@components/Table/TableSeed'
import { Table, TableRow, Theme } from '@material-ui/core'
import { shallow, mount } from 'enzyme'
import 'jest-styled-components'
import React from 'react'
import { createTheme, ThemeProvider } from '@material-ui/core/styles'
import { defaultTheme } from '@styles/theme'

describe('<Table />', () => {
  let theme: Theme

  beforeAll(() => {
    theme = createTheme(defaultTheme)
  })

  it('should render table wrapper', () => {
    const component = shallow(<TableWrapper />)
    expect(component).toMatchSnapshot()
  })

  it('should have table wrapper valid styles', () => {
    const component = mount(
      <ThemeProvider theme={theme}>
        <TableWrapper />
      </ThemeProvider>
    )
    expect(component).toHaveStyleRule('box-shadow', 'none')
  })

  it('should have table scroller valid styles', () => {
    const component = mount(
      <ThemeProvider theme={theme}>
        <TableScroller />
      </ThemeProvider>
    )
    expect(component).toHaveStyleRule('border', `1px solid ${theme.palette.grey[200]}`)
    expect(component).toHaveStyleRule('border-radius', '1rem')
  })

  it('should render table scroller inner table', () => {
    const component = shallow(
      <TableScroller>
        <Table />
      </TableScroller>
    )

    const innerTable = component.find(Table)
    expect(innerTable.exists()).toBeTruthy()
  })

  it('should render table header', () => {
    const component = shallow(
      <ThemeProvider theme={theme}>
        <Table>
          <TableHeader />
        </Table>
      </ThemeProvider>
    )

    expect(component).toMatchSnapshot()
  })

  it('should have table column valid theme styles', () => {
    const component = mount(
      <ThemeProvider theme={theme}>
        <Table>
          <TableHeader>
            <TableRow>
              <TableColumn />
            </TableRow>
          </TableHeader>
        </Table>
      </ThemeProvider>
    )

    // @ts-expect-error
    expect(component.find(TableColumn)).toHaveStyleRule('color', theme.palette.grey[250])
  })

  it('should render table body', () => {
    const component = shallow(
      <Table>
        <TableBody />
      </Table>
    )
    expect(component).toMatchSnapshot()
  })
}

)
