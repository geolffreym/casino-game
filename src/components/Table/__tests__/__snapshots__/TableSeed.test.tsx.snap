// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Table /> should render table body 1`] = `ShallowWrapper {}`;

exports[`<Table /> should render table header 1`] = `ShallowWrapper {}`;

exports[`<Table /> should render table wrapper 1`] = `ShallowWrapper {}`;
